src/http/api/account.ts

```diff
@@ -42,57 +52,52 @@ export function createGetAccountHandler(accountService: AccountService) {
-            return new Response(null, { status: 400 });
+        const defaultAccount = await accountRepository.getBySite(
+            ctx.get('site'),
+        );
+
+        const handle = ctx.req.param('handle');
+        if (handle === 'index') {
+            account = defaultAccount;
+        } else {
+            if (!isHandle(handle)) {
+                return new Response(null, { status: 404 });
+            }
+
+            const apId = await lookupAPIdByHandle(apCtx, handle);
+            if (apId) {
+                account = await accountRepository.getByApId(new URL(apId));
+            }
         }
 
-        const db = ctx.get('db');
         let accountDto: AccountDTO;
 
-        const account = await accountService.getDefaultAccountForSite(site);
-
-        if (!account) {
-            return new Response(null, { status: 404 });
-        }
-
         try {
-            accountDto = {
-                /**
-                 * At the moment we don't have an internal ID for Ghost accounts so
-                 * we use Fediverse ID
-                 */
-                id: account.ap_id,
-                name: account.name || '',
-                handle: getAccountHandle(site.host, account.username),
-                bio: sanitizeHtml(account.bio || ''),
-                url: account.url || '',
-                avatarUrl: account.avatar_url || '',
-                /**
-                 * At the moment we don't support banner images for Ghost accounts
-                 */
-                bannerImageUrl: account.banner_image_url,
-                /**
-                 * At the moment we don't support custom fields for Ghost accounts
-                 */
-                customFields: {},
-                postCount: await accountService.getPostCount(account),
-                likedCount: await accountService.getLikedCount(account),
-                followingCount:
-                    await accountService.getFollowingAccountsCount(account),
-                followerCount:
-                    await accountService.getFollowerAccountsCount(account),
-                /**
-                 * At the moment we only expect to be returning the account for
-                 * the current user, so we can hardcode these values to false as
-                 * the account cannot follow, or be followed by itself
-                 */
-                followsMe: false,
-                followedByMe: false,
-            };
+            //If we found the account in our db and it's an internal account, do an internal lookup
+            if (account?.isInternal) {
+                accountDto = await getAccountDtoFromAccount(
+                    account,
+                    defaultAccount,
+                    accountService,
+                );
+            } else {
+                //Otherwise, do a remote lookup to fetch the updated data
+                accountDto = await getAccountDTOByHandle(
+                    handle || '',
```

Comment:
Why do we need to default to `''`? If the handle is empty we shouldn't even get to this right?